getretake/retake
# feat: move docker compose files to docker dir

**What**
Create a `docker` directory in the root of the repository, and move all docker compose files there. A base file was created which the main, prod and dev compose file extend.

The new files introduced are:
- `docker-compose.base.yml`: The base compose file, includes all of the common services and configurations. It is not meant to be used directly, but extended.
- `docker-compose.dev.yml`: Exposes ports on all containers, builds the `search` and `sync` images locally. Used for local development of Retake.
- `docker-compose.prod.yml`: Doesn't expose ports on any container, introduces Caddy as a reverse proxy to use TLS on the stack, uses the latest images from Dockerhub. Meant to be deployed on an instance as a hobby production deploy.
- `docker-compose.yml`: The main compose file, used for development and testing. Exposes ports on all containers and pulls images from Dockerhub.
- `search.Dockerfile`: Used to build the Retake Search API.
- `sync.Dockerfile`: Used to build the Retake Sync server.

**Why**
Because the current setup is limiting on developers, so it's better to create separate files for each purpose.

**How**
Use a base docker compose file and extend the main, dev and prod compose files. The Dockerfiles were moved too, and ci workflows updated to reflect this change.

**Tests**
None, but this PR can be tested by running on the root of the repository:
1. Run `docker compose -f docker-compose.dev.yml up -d` or any other variation with the available Dockerfiles.
2. Run `docker build -t search -f docker/search.Dockerfile .` to confirm the build context is correct.
